src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -685,369 +685,243 @@ private static bool TryCopyTo(string source, Span<char> destination, out int cha
-            }
+            NumberFormatInfo info = NumberFormatInfo.GetInstance(provider);
 
-            char fmt = ParseFormatSpecifier(format, out int digits);
-            char fmtUpper = (char)(fmt & 0xFFDF); // ensure fmt is upper-cased for purposes of comparison
-            if ((fmtUpper == 'G' && digits < 1) || fmtUpper == 'D')
-            {
-                return value >= 0 ?
-                    TryUInt64ToDecStr((ulong)value, digits, destination, out charsWritten) :
-                    TryNegativeInt64ToDecStr(value, digits, NumberFormatInfo.GetInstance(provider).NegativeSign, destination, out charsWritten);
-            }
-            else if (fmtUpper == 'X')
+            // UInt64NumberBufferLength is the largest number of digits ToNumber can possibly write.
+            byte* pDigits = stackalloc byte[UInt64NumberBufferLength];
+            NumberBuffer number = new NumberBuffer(NumberBufferKind.Integer, pDigits, UInt64NumberBufferLength);
+
+            intFmt.ToNumber(value, ref number);
+
+            if (fmt != 0)
             {
-                // The fmt-(X-A+10) hack has the effect of dictating whether we produce uppercase or lowercase
-                // hex numbers for a-f. 'X' as the fmt code produces uppercase. 'x' as the format code
-                // produces lowercase.
-                return TryInt64ToHexStr(value, (char)(fmt - ('X' - 'A' + 10)), digits, destination, out charsWritten);
+                NumberToString(ref sb, ref number, fmt, digits, info);
             }
             else
             {
-                NumberFormatInfo info = NumberFormatInfo.GetInstance(provider);
-
-                byte* pDigits = stackalloc byte[Int64NumberBufferLength];
-                NumberBuffer number = new NumberBuffer(NumberBufferKind.Integer, pDigits, Int64NumberBufferLength);
-
-                Int64ToNumber(value, ref number);
-
-                char* stackPtr = stackalloc char[CharStackBufferSize];
-                ValueStringBuilder sb = new ValueStringBuilder(new Span<char>(stackPtr, CharStackBufferSize));
-
-                if (fmt != 0)
-                {
-                    NumberToString(ref sb, ref number, fmt, digits, info);
-                }
-                else
-                {
-                    NumberToStringFormat(ref sb, ref number, format, info);
-                }
-                return sb.TryCopyTo(destination, out charsWritten);
+                NumberToStringFormat(ref sb, ref number, format, info);
             }
         }
 
-        public static unsafe string FormatUInt64(ulong value, ReadOnlySpan<char> format, IFormatProvider? provider)
-        {
-            // Fast path for default format
-            if (format.Length == 0)
-            {
-                return UInt64ToDecStr(value, digits: -1);
-            }
+        public static string FormatSByte(sbyte value, ReadOnlySpan<char> format, IFormatProvider? provider)
```

Comment:
The only use here is to avoid making IIntFmt<>, Int32Fmt, FormatInt<,>, and TryFormatInt<,> internal. If you think it's not worthwhile, I'll make the necessary changes.

Reply:
I think it is fine to make IIntFmt internal